src/test/groovy/com/bisnode/opa/client/rest/OpaRestClientSpec.groovy

```diff
@@ -0,0 +1,106 @@
+package com.bisnode.opa.client.rest
+
+import com.bisnode.opa.client.OpaClientException
+import com.bisnode.opa.client.OpaConfiguration
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.github.tomakehurst.wiremock.WireMockServer
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Subject
+
+import java.net.http.HttpClient
+import java.net.http.HttpResponse
+
+import static com.bisnode.opa.client.rest.ContentType.Values.APPLICATION_JSON
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse
+import static com.github.tomakehurst.wiremock.client.WireMock.any
+import static com.github.tomakehurst.wiremock.client.WireMock.anyUrl
+import static com.github.tomakehurst.wiremock.client.WireMock.getRequestedFor
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo
+
+class OpaRestClientSpec extends Specification {
+
+    private static int PORT = 8181
+
+    private String url = "http://localhost:$PORT"
+    @Shared
+    private WireMockServer wireMockServer = new WireMockServer(PORT)
+    @Subject
+    private OpaRestClient client
+
+    def setupSpec() {
+        wireMockServer.start()
+    }
+
+    def cleanupSpec() {
+        wireMockServer.stop()
+    }
+
+    def setup() {
+        client = new OpaRestClient(new OpaConfiguration(url), HttpClient.newHttpClient(), new ObjectMapper())
```

Comment:
I guess it can be inlined in line 29. If not, please remove empty line below.

Reply:
Good practice is to create test subject from scratch for every test case, i will remove the line

Reply:
http://spockframework.org/spock/docs/1.3/spock_primer.html#_fields

```
Instance fields are a good place to store objects belonging to the specification’s fixture. It is good practice to initialize them right at the point of declaration. (Semantically, this is equivalent to initializing them at the very beginning of the setup() method.)
```

Reply:
Didn't know that, ty